assignments/closure.js

```diff
@@ -1,14 +1,25 @@
 // ==== Challenge 1: Write your own closure ====
 // Write a simple closure of your own creation.  Keep it simple!
 
+function sayHello(name) {
+  var text = 'Hello ' + name;
+  var say = function() { 
+    console.log(text); }
+  say();
+}
+sayHello('Carmen'); 
 
 /* STRETCH PROBLEMS, Do not attempt until you have completed all previous tasks for today's project files */
 
 
 // ==== Challenge 2: Create a counter function ====
 const counter = () => {
+  function add() {
+    counter += 1; 
+  }
   // Return a function that when invoked increments and returns a counter variable.
 };
+add(); 
```

Comment:
So you'd want to return a function that increments and returns a new value.
See if you can walk through the logic here:
```js
const counter = (num = 0) => () => ++num;

const iter = counter(); // (defaults to 0 as the initial number)

iter(); // invokes `() => ++num` which implicitly returns the incremented `num` variable.
```